### PR TITLE
Add Tasks functionality

### LINKED ISSUE #8: As a user, I can add a task

### PR DESCRIPTION
**Pull request by kianseong**
### Changes
- Add task model: `Task.java`
- Add task list model: `UniqueTaskList.java`
- Add command to add a task: `AddTaskCommand`
- Add parsers for Tasks: `TaskCommandParser` and `AddTaskCommandParser`
- Add new storage classes to store task data.
- Add tests for new classes.

<br>
Closes #8

### CODE REVIEW

File: src/main/java/seedu/address/MainApp.java [hunk 1]
```diff
@@ -69,28 +74,52 @@ public void init() throws Exception {
...
+
+        try {
+            taskPanelOptional = storage.readTaskPanel();
+            if (taskPanelOptional.isEmpty()) {
+                logger.info("Task data file not found. Will be starting with a sample TaskPanel");
+            }
+            initialTaskPanel = taskPanelOptional.orElseGet(SampleDataUtil::getSampleTaskPanel);
+        } catch (DataConversionException e) {
+            logger.warning("Task data file not in the correct format. Will be starting with an empty TaskPanel");
+            initialTaskPanel = new TaskPanel();
```
Comment (wxy1203) [inline src/main/java/seedu/address/MainApp.java:114]: LGTM

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -143,8 +148,27 @@ public boolean equals(Object obj) {
...
+    public boolean hasTask(Task task) {
+        requireNonNull(task);
+        return taskPanel.hasTask(task);
+    }
+
+    @Override
+    public void addTask(Task task) {
+        taskPanel.addTask(task);
+        // TODO: updateFilteredTaskList
+    }
```
Comment (wxy1203) [inline src/main/java/seedu/address/model/ModelManager.java:173]: LGTM

**Review by wxy1203 — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 3]
```diff
@@ -35,18 +37,25 @@ public LogicManager(Model model, Storage storage) {
...
     }
 
     @Override
     public CommandResult execute(String commandText) throws CommandException, ParseException {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
 
         CommandResult commandResult;
-        Command command = addressBookParser.parseCommand(commandText);
+        Command command;
+        if (commandText.startsWith("task") && Character.isWhitespace(commandText.charAt(4))) {
```
Comment (beetee17) [inline src/main/java/seedu/address/logic/LogicManager.java]: Would using TaskCommand.COMMAND_WORD here be better?

**Review by beetee17 — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/LogicManager.java (hunk 3 above)
Comment (kianseong) [reply src/main/java/seedu/address/logic/LogicManager.java]: Yup, let me update that!

**Pull request merged**